### DISCUSSION TITLE
Error when creating a docker image

### DISCUSSION THREAD

**Discussion by erixschueler** [tsightler/ring-mqtt#740] [Support Q&A]
Thanks first of all for the great work.

Unfortunately, I am encountering the following error when creating the docker container:
`ERROR: failed to solve: failed to register layer: Container ID XXXXXXXXX cannot be mapped to a host ID.`

After a little research I came across the following:
[https://azureossd.github.io/2022/06/30/Docker-User-Namespace-remapping-issues/index.html](https://azureossd.github.io/2022/06/30/Docker-User-Namespace-remapping-issues/index.html)

The mentioned error exists since tag [5.1.3](https://hub.docker.com/layers/tsightler/ring-mqtt/5.1.3/images/sha256-e2258fee3fc8a8964a1e304c41710679c595a38883790a55a1ae56910e2f00fe?context=explore). The previous versions can be built but seems to be outdated. 

Is there a working workaround for this?
Thanks a lot.

**Comment by tsightler** [comment]
Can you please provide more details on exactly what you are trying to do?  What specific steps are you performing to create the docker container?  What platform are you running on?  Usually these types of errors are caused by strange environments, such as trying to run non-root container on Proxmox or something like that, but you didn't really provide any details.  I mean, obviously this doesn't impact most people or the forum would be full of complaints and I couldn't build the docker images that are pushed.

**Reply by erixschueler** [reply]
Thanks first of all for the quick reply.
Of course sorry, I do my best to provide all the details.

OS: `Ubuntu 20.04.6 LTS (GNU/Linux 5.4.0-163-generic x86_64)`

I am using this docker-compose snippet and just run `docker-compose up -d`:

```
version: '2.1'
services:
  ring-mqtt:
    container_name: ring-mqtt
    restart: unless-stopped
    image: tsightler/ring-mqtt:latest
    ports:
      - 8554:8554  
    volumes:
      - /path_to_local_data:/data
    logging:
      options:
        max-size: 10m
        max-file: "3"
```

But the error also appears when I run the following command:

`docker run -it -v /path_to_local_data:/data tsightler/ring-mqtt`

docker version:
```
Client: Docker Engine - Community
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.20.7
 Git commit:        ed223bc
 Built:             Mon Sep  4 12:32:12 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          24.0.6
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.7
  Git commit:       1a79695
  Built:            Mon Sep  4 12:32:12 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.24
  GitCommit:        61f9fd88f79f081d64d6fa3bb1a0dc71ec870523
 runc:
  Version:          1.1.9
  GitCommit:        v1.1.9-0-gccaecfc
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

The docker application is run by a specially created user who has the necessary permissions for this. But it also fails as root.

Locally on my MacBook the image could be built.

I hope this could help you.

**Reply by tsightler** [reply]
I stood up a Ubuntu 20.04 VM and updated it to latest, installed docker and docker-compose from the standard repos and I can't reproduce the behavior you describe.  I don't have the same Docker versions you do however as those versions appear newer than the ones in the official Ubuntu repositories.  This makes me wonder what method you are using to install Docker.  Are you performing a rootless mode Docker install or something like that?
